narwhal/worker/src/handlers.rs

```diff
@@ -170,24 +177,23 @@ impl<V: TransactionValidator> PrimaryToWorker for PrimaryReceiverHandler<V> {
 
         // Keep attempting to retrieve missing batches until we get them all or the client
         // abandons the RPC.
+        // TODO: synchronize() should only be used for header payloads, so remove the broadcast
```

Comment:
What are we replacing async batch fetch for certs with? Going through BatchFetcher? I think if thats the case we would go through BatchFetcher for header payload fetch as well.

Reply:
Header payloads are processed differently from certificate payloads, so to have logic to get them from BatchFetcher, it needs to be properly separated from fetching certificate payloads. Async fetch via BatchFetcher is easier to implement.